File: aspnet-core/src/ProjectManagement.Application/APIs/PMReportProjectContribution/Dto/WeeklyContributionDto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectManagement.APIs.PMReportProjectContribution.Dto
{
    public class WeeklyContributionDto
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public string UserName { get; set; }
        public string UserFullName { get; set; }
        public long ProjectUserBillId { get; set; }
        public long PMReportId { get; set; }
        public string PMReportName { get; set; }
        public long ProjectId { get; set; }
        public string ProjectName { get; set; }
        public byte Contribute { get; set; } = 0;
    }
}

File: angular/src/app/service/model/weekly-contribution.dto.ts
export class WeeklyContributionDto {
  pmReportId?: number;
  pmReportName?: string;
  projectId?: number;
  projectName?: string;
}

export interface ProjectBillDetailDto {
  pmReportId?: number;
  reportName?: string;
  accountName: string;
  billRole: string;
  contribute: number;
  headCount: number;
}

export interface ProjectUserContributionDto {
  projectId: number;
  projectName: string;
  pmName: string;
  totalContribute: number;
  billDetails: ProjectBillDetailDto[];
}

export interface UserGroupContributionDto {
  userId: number;
  userName: string;
  userFullName: string;
  avatarPath: string;
  branchDisplayName: string;
  branchColor: string;
  positionName: string;
  positionColor: string;
  totalHeadCount: number;
  projects: ProjectUserContributionDto[];
}

export interface UserTypeDto {
  displayName: string;
  value: number;
}

export interface ContributionAverageUserDto {
  userId: number;
  userName: string;
  userFullName: string;
  avatarPath: string;
  branchDisplayName: string;
  branchColor: string;
  positionName: string;
  positionColor: string;
  averageContribution: number;
  projects: ProjectUserContributionDto[];
}

export interface ContributionAverageResultDto {
  items: ContributionAverageUserDto[];
  totalCount: number;
  weeklyReportInRange: number;
}